src/chronos/wTraveler/update-user-handler.ts

```diff
@@ -34,24 +58,29 @@ export function handleUpdateUser(event: UpdateUser): void {
     event.logIndex.toString();
 
   const rental = new Rental(rentalId);
-  rental.nftEntity = generateNftId(TYPE, tokenId);
+  rental.nft = generateNftId(TYPE, tokenId);
   rental.borrower = userAccount.id;
   rental.lender = nft.currentOwner;
-  rental.tokenId = tokenId;
-  rental.expirationDate = expires;
+  rental.start_date = event.block.timestamp;
+  rental.expiration_date = expires;
   rental.save();
 
-  nft.currentRental = rentalId;
+  if (expires.gt(event.block.timestamp)) {
+    nft.currentRental = rentalId;
```

Comment:
When updating the `currentRental`, it's necessary first to verify if the rental before the new one is expired. If not expired, then you should set the `expiration_date` of the previous one as `event.block.timestamp`